src/algebra/order/complete_field.lean

```diff
@@ -312,3 +312,26 @@ instance : unique (β ≃+*o γ) := unique_of_subsingleton $ induced_order_ring_
 
 end induced_map
 end linear_ordered_field
+
+section real
+
+/-- Any ring homomorphism `ℝ → ℝ` is an ordered ring isomorphism, and thus there are no
+nontrivial ring homorphism `ℝ → ℝ`. -/
+lemma real.ring_hom_eq_id (f : ℝ →+* ℝ) : f = ring_hom.id ℝ :=
+begin
+  have f_mon : monotone f,
```

Comment:
[Wikipedia](https://en.wikipedia.org/wiki/Real_closed_field#Definitions) mentions the result that every ring homomorphism between real closed fields automatically is order preserving, but we don't have real closed fields in mathlib. However, I think the proof works for any ring_hom between ordered_rings such that every positive element in the domain has a square root (equivalently, the range of the squaring function is the nonnegative elements). Maybe we should extract that as a lemma so it can be used for real closed fields once they are in mathlib?

Reply:
@xroblot what do you think? IMO it is a good idea to extract a sublemma here.

Reply:
Yes, that's a good idea. I'll do that this afternoon.

Reply:
Is `ordered_ring` not enough?

Reply:
@alreadydone ```ordered_ring``` is not enough to prove that there is only one ring homomorphism. One needs also ```archimedean``` to use the fact  that, since the rationals are dense and fixed by the ring homomorphism, the order is preserved. 

I am not sure that was your question though...

Reply:
I mean the `monotone` result. I pushed to the change to [a branch](https://github.com/leanprover-community/mathlib/compare/generalize_real.ring_hom?expand=1) but CI hasn't started running yet...

Reply:
I was a bit wrong; the target needs to be linearly ordered, and `[ordered_ring R] [linear_ordered_ring S]` suffices. [Build has succeeded](https://github.com/leanprover-community/mathlib/compare/generalize_real.ring_hom?expand=1).

Reply:
Great. Thanks for that! I am still not use to try to prove the result in the most generality.

Reply:
Wait. How does that work now? Do I need to do something?

Reply:
You can just open another PR.

Reply:
Ok. thanks